### DISCUSSION TITLE
Allows appDirectory to be set as the root directory

### DISCUSSION THREAD

**Discussion by omakise** [remix-run/remix#7431] [Proposals]
This issue has been resolved using Vite!

**Comment by rossipedia** [comment]
What is the problem you're trying to solve by setting `appDirectory` to the root of the repository?

Most projects that I've seen end up with the structure where all project source code is in a subfolder of the repo (typically `src`, but in Remix apps that's the `app` folder), and top-level things are related to the building and maintaining of that source code. This keeps a clean separation between the application code itself, and the parts of the repo that work _with_ the application code, which often have separate requirements.

My personal opinion: I don't particularly see a benefit to allowing `appDirectory` to be set to `.`, and a bunch of downsides. You have to somehow tell the compiler which parts of the folder structure should be included / excluded.
